textalyze.js

```diff
@@ -1,23 +1,42 @@
 /**
- * This is the base code for v0.1 of our JavaScript text analyzer.
- * Visit https://github.com/jfarmer/text-analysis to see what to do.
- *
- * Send a message in Slack if you're stuck or unsure what to do.  These
- * comments are here to help you, but please delete them as you go along.
+ * The hard coded array used to illustrate the first iteration.
  */
+let arrayToBeCounted = ["a", "a", "a", "b", "b", "c"]
 
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @returns {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  const counts = new Map();
+  return array.reduce(function(map, value) {
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+    if (map.get(value) === undefined) {
+      map.set(value, 1);
+    } else {
+      map.set(value, map.get(value) + 1);
+    }
 
-  return counts;
+    return map;
```

Comment:
You could argue that since `map` is defined locally we can guarantee if `map.get('foo')` is `undefined` then we know for 100% sure that it's never been set.

This is true, but if we're writing code with that knowledge in mind, we could just write:

```javascript
const currentCount = map.get(item) || 0;
return map.set(item, currentCount + 1);
```